datafusion/substrait/src/variation_const.rs

```diff
@@ -38,10 +38,16 @@
 /// The "system-preferred" variation (i.e., no variation).
 pub const DEFAULT_TYPE_VARIATION_REF: u32 = 0;
 pub const UNSIGNED_INTEGER_TYPE_VARIATION_REF: u32 = 1;
+
+#[deprecated(since = "41.0.0", note = "Use `PrecisionTimestamp(Tz)` type instead")]
```

Comment:
SInce we have already released `41` https://crates.io/crates/datafusion/41.0.0 I think we should probably label this with the next version. For example:

```suggestion
#[deprecated(since = "42.0.0", note = "Use `PrecisionTimestamp(Tz)` type instead")]
```

Reply:
Hah yeah, this PR was a long time in the making 😅  fixed in a05637b941199020e1a1d0c995b1bf609250b29c